common/fileformat.h

```diff
@@ -0,0 +1,145 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef COMMON_FILEFORMAT_H
+#define COMMON_FILEFORMAT_H
+
+#include "common/str.h"
+#include "image/bmp.h"
+
+namespace Graphics {
+
+struct Surface;
+
+} // End of namespace Graphics
+
+namespace Common {
+
+namespace FileFormat {
+
+enum Category {
+	kCategoryUnknown,
+
+	kCategoryAudio,
+	kCategoryImage,
+	kCategoryBinary,
+};
+
+enum Lossiness {
+	kLossinessUnknown,
+
+	kLossinessLossless,
+	kLossinessLossy,
+	kLossinessEither,
+};
+
+
+/**
+ * Values representing support for saving or sharing file types on the system.
+ */
+enum FormatSupportLevel {
+	kFormatSupportLevelNone,      ///< Not supported at all
+	kFormatSupportLevelAvoid,     ///< Can be saved/loaded but most likely unsupported even by third-party tools
+	kFormatSupportLevelDefault,   ///< Not natively supported by the operating system, support by third-party tools is either available or unknown
+	kFormatSupportLevelSupported, ///< Well-supported by the operating system
+	kFormatSupportLevelPreferred, ///< Supported by operating system, preferred over other formats if lossless conversion is possible
+};
+
+enum Format {
+	kFormatUnknown,
+
+	// Image formats
+	kBMP,	kFirstImageFormat = kBMP,
+	kPNG,
+	kJPEG,
+	kPICT,	// Macintosh PICT
+
+	// Audio formats
+	kWAV,	kFirstAudioFormat = kWAV,
+	kAIFF,
+
+	// Binary formats
+	kUntypedBinary,	kFirstBinaryFormat = kUntypedBinary,
+
+	// End of list
+	kCount,
+
+	kLastImageFormat = kFirstAudioFormat - 1,
+	kLastAudioFormat = kFirstBinaryFormat - 1,
+	kLastBinaryFormat = kCount - 1,
+};
+
+struct ImageFormatCharacteristics {
+	ImageFormatCharacteristics();
+
+	Lossiness _lossiness;
+	bool _supportAlpha;
+	bool _supportPalette;
+	uint _maxBitDepth;
+};
+
+struct AudioFormatCharacteristics {
+	AudioFormatCharacteristics();
+
+	Lossiness _lossiness;
+};
+
+struct ImageSaveProperties {
+	ImageSaveProperties();
+
+	uint8 _qualityLevel;
+};
+
+typedef bool (*ImageSaveCallback_t)(Common::WriteStream &stream, const Graphics::Surface &surf, const byte *palette, const ImageSaveProperties &saveProperties);
+
+const char *toMimeType(Format format);
+Format fromMimeType(const Common::String &mimeType);
+Format fromFileExtension(const Common::String &extension);
+Category getFormatCategory(Format format);
+
+/**
+ * Returns the default file extension for a format
+ *
+ * @param format              The file format
+ * @param limitTo3Characters  If true, limit the extension to 3 characters
+ * @return 
+*/
+const char *getFormatExtension(Common::FileFormat::Format format, bool limitTo3Characters);
+
+Common::U32String getFormatSaveDescription(Common::FileFormat::Format format);
+
+ImageSaveCallback_t getImageSaveFunction(Format format);
+
+
+bool getImageFormatCharacteristics(Format format, ImageFormatCharacteristics &outCharacteristics);
+bool getAudioFormatCharacteristics(Format format, AudioFormatCharacteristics &outCharacteristics);
```

Comment:
What is the use-case for these audio formats? I do not see it used anywhere

Reply:
Removed